### PR TITLE
Add support for environment variables, annotations, labels and tolera…

### PR DESCRIPTION
**Pull request by blublinsky**
Add support for environment variables, annotations, labels, and tolerations

## Why are these changes needed?

It is quite important to be able to specify environment variables, annotations, labels, and tolerations to control pod's behavior. This is exposed in Kuberay operator but not in the API server

The basic changes are:
1.  Extending cluster and config protos to add new variables and regenerate go client
2.  In the API server project modified podTemplateSpec building in utils cluster to add new values
3.  In the API server project modified template config map creation in utils cluster to add new values
4.  In the API server project modified CRD to API conversion in model converter to create new values

The rest of the changes are due to switching to go 1.19,

## Checks

- [ X] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [X ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by kevin85421 — COMMENTED** [review]
Hi @blublinsky, thank you for the contribution! Could you add tests for this PR? In addition, would you mind explaining why do you move the base image to go 1.19? This may be a breaking change. Thanks!

cc KubeRay API Server folks to review this PR. @Jeffwan @scarlet25151 @wilsonwang371

**Comment by blublinsky** [comment]
Hi @kevin85421 
The PR is so trivial, there is nothing really to test. It just adds standard parameters  to PodSpec, that's it
I am using Go 1.19 for all of my development, 1.17 will work the same.
Also, the API server docker file is in the wrong place - it has to be in the root, but I did not fix it in the PR
Finally development.md refers to the file `docker-image-builder.sh` which does not exist

File: proto/swagger/serve.swagger.json [hunk 1]
```diff
@@ -799,12 +855,16 @@
     "protobufAny": {
       "type": "object",
       "properties": {
-        "@type": {
+        "typeUrl": {
```
Comment (z103cb) [inline proto/swagger/serve.swagger.json:858]: This is breaking api change, is it necessary?

Comment (blublinsky) [reply proto/swagger/serve.swagger.json:858]: This is autogenerated, it is version 1.19 vs 1.17

**Pull request closed without merging**